docs/madr/2-sqlite.md

```diff
@@ -0,0 +1,89 @@
+# Using aos-sqlite Module
+
+- Status: proposed
+- Deciders: [Dylan], [Ariel], [Phil]
+- Date: 2024-07-22
+- Authors: [Atticus]
+
+## Context and Problem Statement
+
+The `aos-sqlite` module is a WebAssembly (Wasm) binary in the Arweave (ao) ecosystem that extends the functionality of the standard `aos` Wasm binary module. The `aos` module includes an `Eval` handler that consumes Lua strings and uses the `load` method of Lua to execute them. The `aos-sqlite` module is compiled with `lua-sqlite3` from C to Wasm, providing additional SQLite functionality. Given that directly using tables has demonstrated scalability issues, the SQLite implementation offers a potential solution. However, due to its novelty in the ecosystem, there are concerns about its stability in larger products. Therefore, using it here for the ANT Registry provides a suitable environment seperate from other products to test its stability.
```

Comment:
> Given that directly using tables has demonstrated scalability issues

This could be explained more clearly, e.g. `Lua tables` to disambiguate with SQLite Tables. The scalability issues could be highlighted explicitly, perhaps with evidence or observations in the wild.

> there are concerns about its stability in larger products.

"Stability" should be quantified here. Version stability? Bugginess? Performance issues?

> Therefore, using it here for the ANT Registry provides a suitable environment seperate from other products to test its stability.

Nit: seperate -> separate.
Also, "testing its stability" isn't the core reason we want it. We want to take advantage of:
- Disk for memory tradeoffs
- Tested (in theory), canonical ways to store, index, search, and access data